include/igl/principal_curvature.cpp

```diff
@@ -900,25 +893,27 @@ IGL_INLINE void igl::principal_curvature(
   // Copy it back
   for (unsigned i=0; i<V.rows(); ++i)
   {
-    PD1.row(i) << cc.curvDir[i][0][0], cc.curvDir[i][0][1], cc.curvDir[i][0][2];
-    PD2.row(i) << cc.curvDir[i][1][0], cc.curvDir[i][1][1], cc.curvDir[i][1][2];
-    PD1.row(i).normalize();
-    PD2.row(i).normalize();
+    if (cc.curv[i].has_value()) {
```

Comment:
This check is new, corresponding to https://github.com/libigl/libigl/pull/2497/files#diff-2419bde26b9d0d869e13e85423962dbf2a8a5349ff94facb26dc150422f5bcebL820.